src/NzbDrone.Common/Extensions/IpAddressExtensions.cs

```diff
@@ -50,7 +50,10 @@ private static bool IsLocalIPv4(byte[] ipv4Bytes)
             // Class C private range: 192.168.0.0 – 192.168.255.255 (192.168.0.0/16)
             bool IsClassC() => ipv4Bytes[0] == 192 && ipv4Bytes[1] == 168;
 
-            return IsLinkLocal() || IsClassA() || IsClassC() || IsClassB();
+            // Tailscale CGNAT range: 100.64.0.0 - 100.127.255.255 (100.64.0.0/10)
+            bool IsTailscaleCGNAT() => ipv4Bytes[0] == 100 && ipv4Bytes[1] >= 64 && ipv4Bytes[1] <= 127;
```

Comment:
Why use a non-static lambda? AFAIK, this would result in an unnecessary heap allocation for the variable capture inside a closure. Using a simple bool local variable would suffice.

A tiny note: maybe check the IP ranges with bit masking? As that feels more natural for IP ranges, and they have been defined based on a binary representation.

More of a general review question, not scoped to this PR particularly, as I can see the previous code is done in the same way.